### PR TITLE
Add image plot, rename existing image plots to heat plots & move plot…

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**
… components into own files

### CODE REVIEW

**Comment by VictoriaBeilsten-Edmands** [comment]
Updated to use NumPy ndarrays

File: api/plot/utils.py [hunk 1]
```diff
@@ -155,14 +162,16 @@ def image(
...
         y: y array
-        domain: array
         title: title of plot
         Returns
         -------
         response: Response
             Response from push_data POST request
         """
-        im = ImageData(key="", values=np.asanyarray(image), domain=domain, **attribs)
+        if "domain" in attribs:
```
Comment (PeterC-DLS) [inline api/plot/utils.py:171]: Check for a rank-3 array with shape [r,c,3] too

File: client/component/PlotComponent.tsx [hunk 2]
```diff
@@ -445,9 +353,22 @@ class PlotComponent extends React.Component<PlotComponentProps, PlotStates> {
       };
       return (
         <>
-          <Plot plotParameters={plotParams} />
+          <Plot plotType={'heat'} plotParameters={plotParams} />
```
Comment (PeterC-DLS) [inline client/component/PlotComponent.tsx:356]: Don't need `{}` for literal value of attributes

File: api/plot/utils.py [hunk 3]
```diff
@@ -155,14 +162,23 @@ def image(
...
+            im = HeatmapData(key="", values=np.asanyarray(image), **attribs)
+        else:
+            try:
+                values = np.asanyarray(image)
+                shape = values.shape
+                assert len(shape) == 3
```
Comment (PeterC-DLS) [inline api/plot/utils.py:177]: Just check (not assert in main code).

File: api/plot/processor.py [hunk 4]
```diff
@@ -77,8 +78,11 @@ def process(
             return self.prepare_new_multiline_data_message(params, plot_config)
         elif message.type == MsgType.new_image_data:
             params = message.params
-            if not isinstance(params, ImageData):
-                params = ImageData(**params)
+            if not (isinstance(params, ImageData) or isinstance(params, HeatmapData)):
```
Comment (PeterC-DLS) [inline api/plot/processor.py:81]: As `HeatmapData` is a subclass of `ImageData` then the second `isinstance` is superfluous

File: api/plot/utils.py [hunk 5]
```diff
@@ -155,14 +162,17 @@ def image(
...
         Returns
         -------
         response: Response
             Response from push_data POST request
         """
-        im = ImageData(key="", values=np.asanyarray(image), domain=domain, **attribs)
+        values = np.asanyarray(image)
+        if "domain" in attribs:
+            im = HeatmapData(key="", values=values, **attribs)
+        elif (len(values.shape) == 3 and values.shape[2] == 3):
```
Comment (PeterC-DLS) [inline api/plot/utils.py:174]: Use `values.ndim == 3`. Also check values is 2D in heatmap case, and if not either case then raise error

File: client/component/PlotComponent.tsx [hunk 6]
```diff
@@ -445,9 +353,22 @@ class PlotComponent extends React.Component<PlotComponentProps, PlotStates> {
       };
       return (
         <>
-          <Plot plotParameters={plotParams} />
+          <Plot plotType='heat' plotParameters={plotParams} />
         </>
       );
+    }
+      else {
```
Comment (PeterC-DLS) [inline client/component/PlotComponent.tsx:360]: Move to end of previous line

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**